lib/tasks/fix_incorrect_hesa_trainee_states.rake

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+namespace :hesa do
+  task :fix_deferred_states do
+    Trainee.deferred.where(record_source: %w[hesa_collection hesa_trn_data]) do |trainee|
```

Comment:
Preloading the related `hesa_student` will save an extra db query per iteration(N+1):

```suggestion
    Trainee.includes(:hesa_student).deferred.where(record_source: %w[hesa_collection hesa_trn_data]) do |trainee|
```